### DISCUSSION TITLE
Multi-site setup with language prefixes on the domain

### DISCUSSION THREAD

**Discussion by alexanderloewe** [craftcms/cms#11337] [General]
I have a multi-site setup with multiple languages for the same domain.

Setup:
`https://domain.com/de/some-slug` (main site)
`https://domain.com/en/some-slug`
`https://domain.com/it/some-slug`

When entering the format `https://domain.com/some-slug`, the content of the main site's entry is output.

I think in this case it would be useful if Craft CMS would not output any of the main site's content to the base URL `https://domain.com/` which is not officially registered.

Or if Craft CMS even handles the redirects. When `https://www.example.com/some-slug` is accessed a redirect should be trigged to `https://www.example.com/prefix/some-slug`

But at least Craft CMS should not give any output to the base URL (https://domain.com/) which has no prefix.

**Comment by brandonkelly** [comment]
This is happening because, when it’s not totally clear which site was requested, Craft will always default to the primary site. If you don’t want to accept any requests that don’t begin with `de/`, `en/` or `it/`, you could enforce that with a server-side redirect. Or you could do it from your base Twig template, with something like:

```twig
{% if not craft.request.fullUri matches '/^(de|en|it)\\//' %}
    {% redirect url(craft.request.fullUri) %}
{% endif %}
